docs/allocators.org

```diff
@@ -346,9 +348,16 @@ This pattern is pretty useful in restricted memory environments such as embedded
 ** TODO C allocator
 The ~C~ standard allocator can also be used, this allocator has high performance but it has less safety feature.
 
+** TODO implementing your own allocator
+
 However, to use this allocator, the ~libC~ is required.
 Adding the ~libC~ in the project will add more dependencies.
 
-
 ** TODO How to use Zig to detect memory leaks
 *** TODO Comparison between gcc-utils sanitizer, Valgrind, and Zig memory leak detection
+
+** TODO ?
+TODO talk about how malloc and calloc work in C with the on demand memory allocation (page fault each time with need more memory) is there a similar system in zig ? can we choose to have on demand memory or can we all allocate at once ?
```

Comment:
not sure to really understand your question, but the professor talked to me about how in c with you allocate memory with malloc it is going to take same time if you allocate 1GB or 1KB. The reason being that it does not really allocate the memory it juste "reserve" it and the allocate on the fly when needed. I was wondering if it worked the same way in Zig or not

Reply:
Ah it's like a copy on write ?

Reply:
if it is what you mean: 

> Copy-on-write (CoW) is a strategy used in computer programming and operating systems to optimize the copying of data. Instead of immediately duplicating the entire data when a copy is requested, the system creates a new reference to the existing data.

then no, maybe it is the same type of optimization yes, but its not this one.

Reply:
Okay this follow the same type of opti